### PR TITLE
remove individual options attributes

### PR DESCRIPTION
**Pull request by vincentsarago**
This PR does:
- remove individual options attributes (nodata, unscale, ...) which are forwarded to the low-level reader and are now replaced with a unique `options={}` 
    ```python
    # before
    with COGReader("cog.tif", nodata=1, resampling_method="bilinear") as cog:
        data = cog.preview()
    # now
    with Reader(COGEO, options={"nodata": 1, "resampling_method": "bilinear"}) as cog:
        data = cog.preview()
    ```

This PR is built on top #534

### CODE REVIEW

**Review by geospatial-jeff — APPROVED** [review]
Nice!  I never liked having to remember which arguments went were.  The one downside of this change is code editors won't know what arguments are available.  It should be documented somewhere what the possible arguments are for 'options', if not already.

File: rio_tiler/io/rasterio.py [hunk 1]
```diff
@@ -75,13 +75,17 @@ class Reader(BaseReader):
...
+    options: reader.Options = attr.ib()
 
     # Context Manager to handle rasterio open/close
     _ctx_stack = attr.ib(init=False, factory=contextlib.ExitStack)
     _minzoom: int = attr.ib(init=False, default=None)
     _maxzoom: int = attr.ib(init=False, default=None)
 
+    @options.default
+    def _options_default(self):
+        return {}
```
Comment (vincentsarago) [inline rio_tiler/io/rasterio.py:87]: I couldn't get mypy to be happy when putting `attr.ib(factory={})` for defaul

File: rio_tiler/reader.py [hunk 2]
```diff
@@ -23,6 +23,16 @@
 from rio_tiler.utils import get_vrt_transform, has_alpha_band, non_alpha_indexes
 
 
+class Options(TypedDict):
```
Comment (vincentsarago) [inline rio_tiler/reader.py:26]: name?

File: rio_tiler/reader.py [hunk 3]
```diff
@@ -23,6 +23,16 @@
...
+    """Reader Options."""
+
+    nodata: Optional[NoData]
```
Comment (geospatial-jeff) [inline rio_tiler/reader.py:29]: I think we want `NotRequired` here isntead https://stackoverflow.com/questions/72326834/how-to-have-optional-keys-in-typeddict

Comment (kylebarron) [reply rio_tiler/reader.py:29]: Also note:

> Note: If you import Required or NotRequired from typing_extensions (on Python <3.11), you should also import TypedDict from typing_extensions.

**Comment by kylebarron** [comment]
> The one downside of this change is code editors won't know what arguments are available

I didn't have time for a close review but I do tend to be wary of this because options dicts or `**kwargs` can feel like black holes at times, where you don't know what values are allowed/disallowed. Maybe with the typeddict it's clear enough

**Pull request merged**